data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -70,25 +77,51 @@ public Long getTimeoutInMilliSec() {
         if (timeoutObj instanceof Long) {
             timeout = (Long) timeoutObj;
         } else if (timeoutObj instanceof String) {
-            timeout = LocalTime.parse((String) timeoutObj).toNanoOfDay() / NANOS_TO_MILLIS;
+            timeout = getMillisFromTimespanString((String) timeoutObj);
         } else if (timeoutObj instanceof Integer) {
             timeout = Long.valueOf((Integer) timeoutObj);
         }
 
         return timeout;
     }
 
+    private long getMillisFromTimespanString(String str) throws TimespanParseException {
+        Matcher matcher = PATTERN.matcher(str);
+        if (!matcher.matches()) {
+            throw new TimespanParseException(str);
+        }
+
+        long millis = 0;
+        String days = matcher.group(1);
+        if (days != null) {
+            millis += Long.parseLong(days) * SECONDS_PER_DAY * 1000L;
+        }
+
+        millis += LocalTime.parse(matcher.group(2)).toNanoOfDay() / 1000000L;
```

Comment:
If working with millis (a duration of time and not a point in time), I'd use Duration over LocalTime.

Reply:
I think it's relatively straightforward to do so.